backend/service/aws/cache.go

```diff
@@ -68,66 +68,65 @@ func (c *client) processAllAutoScalingGroups(ctx context.Context, client *region
 	c.log.Info("starting to process auto scaling groups for region", zap.String("region", client.region))
 	// 100 is the maximum amount of records per page allowed for this API
 	input := autoscaling.DescribeAutoScalingGroupsInput{
-		MaxRecords: aws.Int64(100),
+		MaxRecords: aws.Int32(100),
 	}
 
-	err := client.autoscaling.DescribeAutoScalingGroupsPagesWithContext(ctx, &input,
-		func(page *autoscaling.DescribeAutoScalingGroupsOutput, lastPage bool) bool {
-			for _, asg := range page.AutoScalingGroups {
-				protoAsg, err := ptypes.MarshalAny(newProtoForAutoscalingGroup(asg))
-				if err != nil {
-					c.log.Error("unable to marshal asg proto", zap.Error(err))
-					continue
-				}
+	paginator := autoscaling.NewDescribeAutoScalingGroupsPaginator(client.autoscaling, &input)
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			c.log.Error("unable to get next asg page", zap.Error(err))
+			continue
```

Comment:
this feels like it should be a break, otherwise i think it will keep repeating the request, which could be broken